market-generator/src/main/java/com/market/banica/generator/util/SnapshotPersistence.java

```diff
@@ -0,0 +1,75 @@
+package com.market.banica.generator.util;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.market.banica.common.util.ApplicationDirectoryUtil;
+import com.market.banica.generator.model.MarketTick;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class SnapshotPersistence {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotPersistence.class);
+
+    private static final String DATABASE_FILE_NAME = "tickDatabase.dat";
+
+    private static final Kryo kryoHandle = new Kryo();
+
+    public SnapshotPersistence() {
+        initKryo();
+    }
+
+    public void persistTicks(Map<String, Set<MarketTick>> newTicks) throws FileNotFoundException {
```

Comment:
why is it Map, but not Set ?

Reply:
I talked with Nikola and he preferred to be Map as he would be using this utility in his MarketState class.
There are both plusses and minuses for both Map of Sets and Set of every tick.
If you believe Set would be better, I can change it.